src/pages/Home/components/TodoCreator/TodoCreator.tsx

```diff
@@ -1,66 +1,98 @@
+// import { ChangeEvent, useState } from 'react';
 
+import { useForm } from 'react-hook-form';
 import { useDispatch, useSelector } from 'react-redux';
 
 import Button from '@mui/material/Button';
 import Typography from '@mui/material/Typography';
 import CircularProgress from '@mui/material/CircularProgress';
+import Tooltip from '@mui/material/Tooltip';
 
 import { AppDispatch, RootState } from 'src/redux/store';
 import { createTask } from 'src/redux/tasks';
 
 // import { useTodoContext } from '../../context/Todo';
 
 import { StyledOutlinedInput } from './styled';
+
+type FormData = {
+  task: string;
+};
+
 export default function TodoCreator() {
   // const { createTodo } = useTodoContext();
   const dispatch = useDispatch<AppDispatch>();
   const { isCreating } = useSelector((state: RootState) => state.tasks);
 
-  const [text, setText] = useState('');
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    resetField,
+  } = useForm<FormData>();
 
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    setText(event.target.value);
+  const onSubmit = (data: FormData) => {
+    dispatch(createTask(data.task)).then(() => {
+      resetField('task');
+    });
   };
 
-  const handleCreate = () => {
-    if (text.trim() === '') return;
-
-    // createTodo(text);
-    dispatch(createTask(text));
-    setText('');
-  };
+  // createTodo(text);
+  // console.log(errors.task);
 
   return (
-    <StyledOutlinedInput
-      fullWidth
-      placeholder="Create your todo"
-      endAdornment={
-        <Button
-          disableElevation
-          variant="contained"
-          disabled={isCreating}
-          onClick={handleCreate}
-        >
-          {isCreating && (
-            <CircularProgress
-              size={25}
-              style={{
-                minWidth: '60px',
-              }}
-              sx={{
-                color: 'white',
-                display: 'flex',
-                justifyContent: 'center',
-                textAlign: 'center',
-              }}
-            />
-          )}
-          {!isCreating && <Typography>Create</Typography>}
-        </Button>
-      }
-      value={text}
-      onChange={handleChange}
-    />
+    <form onSubmit={handleSubmit(onSubmit)}>
+      <Tooltip
+        title={errors.task?.message}
+        open={!!errors.task}
+        disableHoverListener={true}
+      >
+        <StyledOutlinedInput
+          {...register('task', {
+            required: {
+              value: true,
+              message: 'This field is required',
+            },
+            minLength: {
+              value: 3,
+              message: 'Should not be less than 3 symbols',
+            },
+            maxLength: {
+              value: 30,
+              message: 'Should not be larger than 30 symbols',
+            },
+          })}
+          fullWidth
+          error={!!errors.task}
+          type="text"
+          id="outlined-error"
```

Comment:
What is this id? Why it is here and why it has that name? How it is relevant to `task` and `create task`?